gateway/gateway_test.go

```diff
@@ -414,11 +414,15 @@ var _ = Describe("Gateway", func() {
 				err = gateway.StartWebHandler(ctx)
 				Expect(err).To(BeNil())
 			}()
-			getEndpoing, postEndpoints, deleteEndpoints := getEndpointMethodMap()
-
-			verifyEndpoing(getEndpoing, http.MethodGet)
-			verifyEndpoing(postEndpoints, http.MethodPost)
-			verifyEndpoing(deleteEndpoints, http.MethodDelete)
+			require.Eventually(GinkgoT(), func() bool {
+				verifyEndpoint([]string{"/version"}, http.MethodGet)
```

Comment:
We don't want the test to fail (test execution to halt with failure), while the `Eventually` block is being executed.
Furthermore, since this test is already using gomega, it would be better to use gomega's Eventually instead.

Reply:
modified.

Reply:
we are still making assertions inside the eventually block, right?

Reply:
fixed.